sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonJsonSerializer.java

```diff
@@ -82,36 +84,70 @@ public String convertMemberName(Member member) {
         if (Modifier.isTransient(member.getModifiers())) {
             return null;
         }
+
+        VisibilityChecker<?> visibilityChecker = mapper.getVisibilityChecker();
         if (member instanceof Field) {
             Field f = (Field) member;
-            if (f.isAnnotationPresent(JsonIgnore.class)) {
+
+            if (f.isAnnotationPresent(JsonIgnore.class) || !visibilityChecker.isFieldVisible(f)) {
                 return null;
             }
+
             if (f.isAnnotationPresent(JsonProperty.class)) {
                 String propertyName = f.getDeclaredAnnotation(JsonProperty.class).value();
                 return CoreUtils.isNullOrEmpty(propertyName) ? f.getName() : propertyName;
             }
+
             return f.getName();
         }
 
         if (member instanceof Method) {
             Method m = (Method) member;
-            String methodNameWithoutJavaBeans = removePrefix(m);
-            if (m.isAnnotationPresent(JsonIgnore.class)) {
+
+            /*
+             * If the method isn't a getter, is annotated with JsonIgnore, or isn't visible to the ObjectMapper ignore
+             * it.
+             */
+            if (!verifyGetter(m)
+                || m.isAnnotationPresent(JsonIgnore.class)
+                || !visibilityChecker.isGetterVisible(m)) {
```

Comment:
It would be useful log a warning if the method has any annotation but is non-public. It will let the users know that the visibility of the method is the reason for ignoring it. Same for field too.

Reply:
Only log in the instance of the method/field being ignored due to visibility and not due to annotations?